Sources/TensorFlow/Layer.swift

```diff
@@ -56,7 +83,7 @@ public extension Layer {
     @usableFromInline
     internal func _vjpInferring(from input: Input)
         -> (value: Output, pullback: (Output.TangentVector)
-            -> (TangentVector, Input.TangentVector)) {
+            -> (AllDifferentiableVariables, Input.TangentVector)) {
```

Comment:
Could you please revert this change? `TangentVector` is a correct type (though `AllDifferentiableVariables` is constrained to be equal to it) and will be the only correct type after we deprecate `AllDifferentiableVariables`.

Reply:
I changed this because if I don't I get this error:
```
/Users/eaplatanios/Development/GitHub/swift-apis/Sources/TensorFlow/Layer.swift:82:6: error: function result's 'pullback' type does not match 'inferring(from:)'
    @differentiating(inferring(from:))
     ^
/Users/eaplatanios/Development/GitHub/swift-apis/Sources/TensorFlow/Layer.swift:85:38: note: 'pullback' does not have expected type '(Self.Output.TangentVector) -> (Self.AllDifferentiableVariables, Self.Input.TangentVector)'
        -> (value: Output, pullback: (Output.TangentVector)
                                     ^~~~~~~~~~~~~~~~~~~~~~
/Users/eaplatanios/Development/GitHub/swift-apis/Sources/TensorFlow/Layer.swift:77:10: note: 'inferring(from:)' defined here
    func inferring(from input: Input) -> Output {
         ^
```

Reply:
Hmm, that looks like a bug. I guess we can deal with it later.

Reply:
Yeah I assumed it was a bug, but didn't look into it too much because I was aware that `AllDifferentiableVariables` is going away soon. :)